extensions/binding/org.eclipse.smarthome.binding.astro.test/src/test/groovy/org/eclipse/smarthome/binding/astro/discovery/test/AstroDiscoveryTest.groovy

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.astro.discovery.test
+
+import static org.hamcrest.CoreMatchers.*
+import static org.junit.Assert.*
+
+import org.eclipse.smarthome.config.core.Configuration
+import org.eclipse.smarthome.config.discovery.DiscoveryListener
+import org.eclipse.smarthome.config.discovery.DiscoveryResult
+import org.eclipse.smarthome.config.discovery.DiscoveryService
+import org.eclipse.smarthome.core.thing.ThingTypeUID
+import org.eclipse.smarthome.core.thing.ThingUID
+import org.junit.Before
+import org.junit.Test
+import org.eclipse.smarthome.binding.astro.AstroBindingConstants
+import org.eclipse.smarthome.binding.astro.discovery.AstroDiscoveryService
+import org.eclipse.smarthome.binding.astro.test.AstroOSGiTest
+
+/**
+ * OSGi tests for the {@link AstroDiscoveryService}
+ *
+ * @author Petar Valchev - Initial implementation
+ * @author Svilen Valkanov - Replaced header
+ */
+class AstroDiscoveryTest extends AstroOSGiTest {
+    AstroDiscoveryService discoveryService
+    // a listener for discovery events
+    DiscoveryListenerMock discoveryListenerMock
+
+    @Before
+    public void setUp(){
+        discoveryListenerMock = new DiscoveryListenerMock()
+        registerService(discoveryListenerMock, DiscoveryListener.class.getName())
+
+        discoveryService = getService(DiscoveryService, AstroDiscoveryService)
+        assertThat "Could not get DiscoveryService",
+                discoveryService,
+                is(notNullValue())
+
+        discoveryService.addDiscoveryListener(discoveryListenerMock)
+    }
+
+    @Test
+    public void 'sun thing is discovered'(){
+        assertDiscoveredThing(TEST_SUN_THING_ID)
+    }
+
+    @Test
+    public void 'moon thing is discovered'(){
+        assertDiscoveredThing(TEST_MOON_THING_ID)
+    }
+
+    private void assertDiscoveredThing(String thingID){
+        discoveryService.startScan()
+
+        waitForAssert({
+            switch(thingID) {
+                case (TEST_SUN_THING_ID) :
+                    assertThat "Sun thing was not discovered",
+                    discoveryListenerMock.isSunThingDiscovered,
+                    is(true)
+                    break
+                case(TEST_MOON_THING_ID) :
+                    assertThat "Moon thing was not discovered",
+                    discoveryListenerMock.isSunThingDiscovered,
+                    is(true)
+                    break
+            }
+        })
+    }
+
+    public class DiscoveryListenerMock implements DiscoveryListener{
```

Comment:
You have two pretty powerful mocking facilities:
* You have defined a dependency to mockito
* You are still using groovy

But still decided to implemented the mock by yourself. Why?